### DISCUSSION TITLE
Clarinet fingering variant?

### DISCUSSION THREAD

**Discussion by thouldcroft** [cardonabits/haxo-hw#13] [Ideas]
Saxophone and clarinet fingerings are CLOSE but not identical. I haven’t throughly gone through the design (I just discovered this project today) but would it be feasible to create a pcb variant for clarinet fingering? I would try to have a go at it but pcb design is a field  I’m familiar with but far from fluent in.

**Comment by jcard0na** [comment]
Hi @thouldcroft ,

I would start by trying to see if the current PCB with different key mappings might work for a clarinet player.  I only tried to play the clarinet for some time, a long time ago.  I remember that the octave key that raises 1 8ve on the sax goes up a 12ve on the clarinet.  This is easy to implement in software.
Other fingerings can be remapped as well by altering the [note mapping file](https://github.com/jcard0na/haxo-rs/blob/main/notemap.json).  If you can get 90% there just with software changes, you'll have saved yourself 99% of the effort.

However, if you actually do need to modify the PCB, go for it.  I've released all the files that you need to replicate/fork the project.  You don't really need a lot of skill to change the layout; the PCB is so large that the is ample room for laying out traces.  The challenges will be working out the ergonomics.  I went through many iterations to get something that feels comfortable (to me!).  

That said, probably the biggest challenge though would be the naming... the hackrinet :stuck_out_tongue_winking_eye: ?
